### DISCUSSION TITLE
Feature Request: Enable and Disable User Scripts

### DISCUSSION THREAD

**Discussion by msmcconnell** [foxglove/community#992] [General]
User Scripts are in my opinion the single best feature of foxglove. They allow me to quickly extract subfields of messages and manipulate the data without having to first dump rosbag data. This dramatically reduces my cycle time. 
However, there is no option to turn certain scripts on or off. The only option is to delete them wholesale or edit the script to make it not function. This makes these really useful scripts much harder to maintain. I'm requesting a simple toggle on each script to enable or disable it.

**Comment by defunctzombie** [comment]
A script is only enabled if its output topics are subscribed to by a panel. That acts as a sort of implicit "enable/disable" for a script. If no panel subscribes the script does not run. Can you elaborate a bit more on your workflow and how it is different from that behavior?

**Reply by msmcconnell** [reply]
Ah I didn't realize that. I've had some performance issues with uncompressed video in mcap files going into a bunch of different scripts due to the need to publish the results in multiple formats. Wanted to toggle off some of them, but removing them from the display list certainly makes sense as well. Maybe this isn't needed then.
